src/unit_tests/wazuh_db/test_wdb_parser.c

```diff
@@ -740,7 +741,479 @@ void test_wdb_parse_rootcheck_save_update_insert_success(void **state)
     os_free(query);
 }
 
-/* vuln_cves Tests */
+/* Tests osinfo */
+
+void test_osinfo_syntax_error(void **state) {
+    int ret = -1;
+    test_struct_t *data  = (test_struct_t *)*state;
+    char *query = NULL;
+
+    os_strdup("agent 000 osinfo", query);
+
+    expect_value(__wrap_wdb_open_agent2, agent_id, atoi(data->wdb->id));
+    will_return(__wrap_wdb_open_agent2, (wdb_t*)1); // Returning any value
+    expect_string(__wrap__mdebug2, formatted_msg, "Agent 000 query: osinfo");
+
+    expect_string(__wrap__mdebug1, formatted_msg, "DB(000) Invalid DB query syntax.");
+    expect_string(__wrap__mdebug2, formatted_msg, "DB(000) query error near: osinfo");
+
+    ret = wdb_parse(query, data->output);
+
+    assert_string_equal(data->output, "err Invalid DB query syntax, near 'osinfo'");
+    assert_int_equal(ret, OS_INVALID);
+
+    os_free(query);
+}
+
+void test_osinfo_invalid_action(void **state) {
+    int ret = -1;
+    test_struct_t *data  = (test_struct_t *)*state;
+    char *query = NULL;
+
+    os_strdup("agent 000 osinfo invalid", query);
+    expect_value(__wrap_wdb_open_agent2, agent_id, atoi(data->wdb->id));
+    will_return(__wrap_wdb_open_agent2, (wdb_t*)1); // Returning any value
+    expect_string(__wrap__mdebug2, formatted_msg, "Agent 000 query: osinfo invalid");
+
+    ret = wdb_parse(query, data->output);
+
+    assert_string_equal(data->output, "err Invalid osinfo action: invalid");
+    assert_int_equal(ret, OS_INVALID);
+
+    os_free(query);
+}
+
+void test_osinfo_missing_action(void **state) {
+    int ret = -1;
+    test_struct_t *data  = (test_struct_t *)*state;
+    char *query = NULL;
+
+    os_strdup("", query);
+
+    ret = wdb_parse_osinfo(data->wdb, query, data->output);
+
+    assert_string_equal(data->output, "err Missing osinfo action");
+    assert_int_equal(ret, OS_INVALID);
+
+    os_free(query);
+}
+
+void test_osinfo_get_error(void **state) {
+    int ret = OS_INVALID;
+    test_struct_t *data  = (test_struct_t *)*state;
+    char *query = NULL;
+
+    os_strdup("get", query);
+
+    // wdb_agents_get_sys_osinfo
+    will_return(__wrap_wdb_agents_get_sys_osinfo, NULL);
+    will_return_count(__wrap_sqlite3_errmsg, "ERROR MESSAGE", -1);
+
+    ret = wdb_parse_osinfo(data->wdb, query, data->output);
+
+    assert_string_equal(data->output, "err Cannot get sys_osinfo database table information; SQL err: ERROR MESSAGE");
+    assert_int_equal(ret, OS_INVALID);
+
+    os_free(query);
+}
+
+void test_osinfo_get_success(void **state) {
+    int ret = OS_INVALID;
+    test_struct_t *data  = (test_struct_t *)*state;
+    char *query = NULL;
+    char *result = NULL;
+
+    os_strdup("get", query);
+    os_strdup("{}", result);
```

Comment:
Because the response from wazuhDB methods are json arrays and an empty response would be **[]** maybe it will be more standard to replace **"{}"** with **"[]"**

Reply:
Totally right! Fixed!